### PR TITLE
balance chunks in CI workflow

### PR DESCRIPTION
**Pull request by bernt-matthias**
fixes: https://github.com/galaxyproject/tools-iuc/issues/2959

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -80,27 +81,24 @@ jobs:
...
                    PLANEMO_OPTIONS="--biocontainers --no_dependency_resolution --no_conda_auto_init"
                fi
-               PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$DIR"/tool_test_output.json "$TESTPATH" || true
+               json=$(mktemp -p json_output --suff .json)
+               PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE --biocontainers --no_dependency_resolution --galaxy_python_version ${{ matrix.python-version }} --test_output_json $json "$TOOLS" || true
                docker system prune --all --force --volumes || true
            done < tool.list
     - name: Merge tool_test_output.json files
-      run: find . -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
+      run: planemo merge_test_reports json_output/*json tool_test_output.json
```
Comment (nsoranzo) [inline .github/workflows/ci.yaml:101]:
```suggestion
      run: planemo merge_test_reports json_output/*.json tool_test_output.json
```

**Comment by bernt-matthias** [comment]
- 1h 28m 4s now .. 42 test(s) out of 3055.
- 3h 11m 15s before .. 36 test(s) out of 3171.

Happy about the time diff, but I do not understand the diff in the number of tests ... ?

**Comment by mvdbeek** [comment]
Maybe the chunking logic isn't quote right? Can you compare the tool ids to see what is missing ?

**Comment by bernt-matthias** [comment]
Lets run the workflows on the same commit... Also I had to re-run and this led to test included multiple times.

**Comment by mvdbeek** [comment]
3179 tests now 😌

**Comment by bernt-matthias** [comment]
Yeah, odd. I started to look in the json files. Considering the test ids I find duplicated entries in the results of the original version of the workflow. In the current run essentially all from the samtools folder:

e.g.

```
"functional.test_toolbox.TestForTool_samtools_view.test_tool_000021"
"functional.test_toolbox.TestForTool_samtools_view.test_tool_000021" 
```

also in the results from the original version of the workflow two tests are missing:

```
"functional.test_toolbox.TestForTool_data_manager_snpeff_databases.test_tool_000000"
"functional.test_toolbox.TestForTool_data_manager_snpeff_download.test_tool_000000"
```

**Comment by mvdbeek** [comment]
So we're testing more tools now (or we have more test results) ?
As long as the current run doesn't skip any tests I think this is ok (a little weird too, but ok)

**Comment by bernt-matthias** [comment]
Seems that the current workflow skips the data_manager_snpeff_databases tool. At least I could not find it in the merged json / html.

**Comment by bernt-matthias** [comment]
And the results contain all tests of samtools twice. So there is a pattern. Maybe something with the shed.yml files?

**Comment by bernt-matthias** [comment]
ci_find_repos searches for shed.yml files. For samtools there is one in the main dir (aiming at defining the suite) and one per tool. 

No idea how to fix. Suggestions?

Not sure why data_manager_snpeff_databases is missing in the old tests .. could not spot a problem in the yml file.

**Comment by mvdbeek** [comment]
> No idea how to fix. Suggestions?

Merge this PR ? ;)

**Comment by bernt-matthias** [comment]
Sounds like a good idea. Was just wondering if this loop really gets all tools https://github.com/galaxyproject/planemo/blob/57e26f5ecf880a552cb956f3e3287995d4c2aead/planemo/commands/cmd_ci_find_tools.py#L31 .. or maybe skip invalid / malformed tools ... but I guess this is tested

**Comment by mvdbeek** [comment]
That could well be. Happy to merge this if you restore the schedule.

**Comment by bernt-matthias** [comment]
Squashed and restored the `on` statement to weekly

**Comment by mvdbeek** [comment]
Amazing, thanks for figuring this out!

**Pull request merged**